bf2fc6cc711aee1a0c2a/kas-fleet-manager
# Add new kafkafields largerkafka functionality

## Description

Related to https://issues.redhat.com/browse/MGDSTRM-8385.

Adds the following fields in the response returned by the getInstanceTypesByCloudProviderAndRegion endpoint of the instance types api. Specifically, on the instance size part of the response.
* `max_message_size`. an object that contains a `bytes` attribute. The format is in bytes. This follows the same format as other fields available in the response like `ingress_throughput_per_sec`. The data type for the `bytes` attribute is an int64
* `supported_az_modes`. a list that can contain `single` or `multi` (or both) as its values returned
* `display_name`. string.
* `min_in_sync_replicas`. int32.
* `replication_factor`. int32.
* `lifespan_seconds`. int32. nullable. It is not returned if null or not set

Adds the following fields in the `config/kafka-instance-types-configuration.yaml` file:
* `maxMessageSize`. string. The format is a K8s Quantity
* `supportedAZModes`. list of string. The values of the list can be `single`, `multi`(or both).
* `displayName`. string.
* `minInSyncReplicas`. int.
* `replicationFactor`. int.
* `lifespanSeconds`. *int. If not set or set to nil the instance type is considered to not have an expiration time

Additionally, to fix an accuracy issue the data type of the following already existing fields has been changed from float32 to int64:
* `ingress_throughput_per_sec`
* `egress_throughput_per_sec`
* `max_data_retention_size`

## Verification Steps

* Verify unit tests and integration tests pass

* Run kas fleet manager and perform a request to the /api/kafkas_mgmt/v1/instance_types/aws/us-east-1 endpoint. Verify that the following new fields appear and they appear with the values configured in the config file:
```
min_in_sync_replicas
replication_factor
supported_az_nodes
max_message_size (an object with a bytes attribute)
display_name (on the kafka size object)
lifespan_seconds. This attribute should only appear for developer instance type.
```

* Verify that the values in bytes reported for the following fields are accurate and there are no rounding errors compared to what's been specified in the config file:
```
max_message_size
ingress_throughput_per_sec
egress_throughput_per_sec
max_data_retention_size
```
* Verify that the newly configured attributes in config/kafka-instance-types-configuration.yaml match the kafka instance limits defined in the spreadsheet


## Checklist (Definition of Done)
